drivers/ethernet/Kconfig.stm32_hal

```diff
@@ -88,4 +88,24 @@ config ETH_STM32_CARRIER_CHECK_RX_IDLE_TIMEOUT_MS
 	  Set the RX idle timeout period in milliseconds after which the
 	  PHY's carrier status is re-evaluated.
 
+config ETH_STM32_AUTO_NEGOTIATION_ENABLE
+	bool "Enable autonegotiation mode"
+	default y
+	help
+	  Enable this if using autonegotiation
+
```

Comment:
You should add:
`if ETH_STM32_AUTO_NEGOTIATION_ENABLE!=y`
so that next switches are present only when ETH_STM32_AUTO_NEGOTIATION_ENABLE is disabled

Reply:
cool, added this below!